tests/ci/integration/ibmtpm_patch/ibmtpm-mainline-awslc.patch

```diff
@@ -0,0 +1,122 @@
+  Error:
+@@ -127,7 +133,7 @@ BIGNUM* BigInitialized(BIGNUM* toInit, bigConst initializer)
+ 	return NULL;
+     toInit->d     = (BN_ULONG*)&initializer->d[0];
+     toInit->dmax  = (int)initializer->allocated;
+-    toInit->top   = (int)initializer->size;
++    BN_FIELD_SIZE(toInit)   = (int)initializer->size;
+     toInit->neg   = 0;
+     toInit->flags = 0;
+     return toInit;
+@@ -156,7 +162,7 @@ static void BIGNUM_print(const char* label, const BIGNUM* a, BOOL eol)
+ 	}
+     if(a->neg)
+ 	printf("-");
+-    for(i = a->top, d = &a->d[i - 1]; i > 0; i--)
++    for(i = BN_FIELD_SIZE(a), d = &a->d[i - 1]; i > 0; i--)
+ 	{
+ 	    int      j;
+ 	    BN_ULONG l = *d--;
+@@ -209,7 +215,7 @@ BOOL BnMathLibraryCompatibilityCheck(void)
+     // Convert the test data to an OpenSSL BIGNUM
+     BN_bin2bn(test, sizeof(test), osslTemp);
+     // Make sure the values are consistent
+-    GOTO_ERROR_UNLESS(osslTemp->top == (int)tpmTemp->size);
++    GOTO_ERROR_UNLESS(BN_FIELD_SIZE(osslTemp) == (int)tpmTemp->size);
+     for(i = 0; i < tpmTemp->size; i++)
+ 	GOTO_ERROR_UNLESS(osslTemp->d[i] == tpmTemp->d[i]);
+     OSSL_LEAVE();
+@@ -558,6 +564,18 @@ LIB_EXPORT BOOL BnEccModMult2(bigPoint            R,  // OUT: computed point
+ 	EC_POINT_mul(E->G, pR, bnD, pQ, bnU, E->CTX);
+     else
+ 	{
++#ifdef OPENSSL_IS_AWSLC
++        EC_POINT *pR1 = EC_POINT_new(E->G);
++        EC_POINT *pR2 = EC_POINT_new(E->G);
++        int OK;
++
++        OK = EC_POINT_mul(E->G, pR1, NULL, pS, bnD, E->CTX);
++        OK &= EC_POINT_mul(E->G, pR2, NULL, pQ, bnU, E->CTX);
++        OK &= EC_POINT_add(E->G, pR, pR1, pR2, E->CTX);
++
++        EC_POINT_free(pR1);
++        EC_POINT_free(pR2);
++#else
+ 	    const EC_POINT* points[2];
+ 	    const BIGNUM*   scalars[2];
+ 	    points[0]  = pS;
+@@ -565,6 +583,7 @@ LIB_EXPORT BOOL BnEccModMult2(bigPoint            R,  // OUT: computed point
+ 	    scalars[0] = bnD;
+ 	    scalars[1] = bnU;
+ 	    EC_POINTs_mul(E->G, pR, NULL, 2, points, scalars, E->CTX);
++#endif
+ 	}
+     PointFromOssl(R, pR, E);
+     EC_POINT_free(pR);
+diff --git a/src/BnToOsslMath.h b/src/BnToOsslMath.h
+index 165304d..78f295b 100644
+--- a/src/BnToOsslMath.h
++++ b/src/BnToOsslMath.h
+@@ -81,7 +81,7 @@
+ // Check the bignum_st definition against the one below and either update the
+ // version check or provide the new definition for this version.
+ #  error Untested OpenSSL version
+-#elif OPENSSL_VERSION_NUMBER >= 0x10100000L
++#elif OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(OPENSSL_IS_AWSLC)
+ // from crypto/bn/bn_lcl.h (OpenSSL 1.x) or crypto/bn/bn_local.h (OpenSSL 3.0)
+ struct bignum_st
+ {
+diff --git a/src/TpmProfile_Common.h b/src/TpmProfile_Common.h
+index 13dfc0d..3c87c3e 100644
+--- a/src/TpmProfile_Common.h
++++ b/src/TpmProfile_Common.h
+@@ -65,6 +65,8 @@
+ // Requires basic YES/NO defines are already set (by TpmBuildSwitches.h)
+ // Less frequently changed items are in other TpmProfile Headers.
+
++#include <openssl/base.h>
```

Comment:
This is an AWS-LC specific header, if you just need access to OPENSSL_NO_CAMELLIA this should use opensslconf.h which is common with OpenSSL.